src/hardware/adlib.cpp

```diff
@@ -74,14 +80,22 @@ struct Handler : public Adlib::Handler {
 		adlib_write_index(port, val);
 		return opl_index;
 	}
-	virtual void Generate(mixer_channel_t &chan, const uint16_t samples)
+	virtual void Generate(mixer_channel_t &chan, const uint16_t frames)
 	{
-		int16_t buf[1024 * 2];
-		int remaining = samples;
+		int16_t buf[render_frames * 2];
+		float float_buf[render_frames * 2];
```

Comment:
Same thought; if we had `buf` and `float_buf` as members in the Handler, could use them in all the Generate function. I would suggest tightening them down too; say 256 or 128 elements.

Reply:
Yeah the size could be 256.